examples/seismic/elastic/wavesolver.py

```diff
@@ -0,0 +1,93 @@
+from devito import TimeFunction, memoized_meth, error
+from examples.seismic import Receiver
+from examples.seismic.elastic.operators import ForwardOperator
+
+
+class ElasticWaveSolver(object):
+    """
+    Solver object that provides operators for seismic inversion problems
+    and encapsulates the time and space discretization for a given problem
+    setup.
+
+    :param model: Physical model with domain parameters
+    :param source: Sparse point symbol providing the injected wave
+    :param receiver: Sparse point symbol describing an array of receivers
+    :param space_order: Order of the spatial stencil discretisation (default: 4)
+
+    Note: This is an experimental staggered grid elastic modeling kernel.
+    Only 2D supported
+    """
+    def __init__(self, model, source, receiver, space_order=4, **kwargs):
+        self.model = model
+        self.source = source
+        self.receiver = receiver
+
+        self.space_order = space_order
+        # Time step can be \sqrt{3}=1.73 bigger with 4th order
+        self.dt = self.model.critical_dt
+        # Cache compiler options
+        self._kwargs = kwargs
+        if model.grid.dim != 2:
+            error("This is an experimental staggered grid elastic modeling kernel." +
```

Comment:
rather than ``error``, which lets the execution proceeds, I'd say ``raise ValueError("...")``, or perhaps even better ``raise NotImplementedError("...")``

Reply:
Was following how it is in the examples but can change

Reply:
``error + raise`` is something we inherited from initial implementations and I don't see how it can be useful . Let's just use ``raise``, and let's change this pattern incrementally wherever we spot it. Unless I'm missing something...

Reply:
I'' l be on the lookout